File: chat-service/src/main/java/com/yazikochesalna/chatservice/config/OpenApiConfig.java
package com.yazikochesalna.chatservice.config;

import io.swagger.v3.oas.annotations.OpenAPIDefinition;
import io.swagger.v3.oas.annotations.info.Info;
import io.swagger.v3.oas.annotations.security.SecurityScheme;
import io.swagger.v3.oas.annotations.security.SecuritySchemes;
import io.swagger.v3.oas.annotations.enums.SecuritySchemeType;
import org.springframework.context.annotation.Configuration;
import io.swagger.v3.oas.annotations.enums.SecuritySchemeIn;

@Configuration
@OpenAPIDefinition(info = @Info(title = "Chat-service", version = "v1"))
@SecuritySchemes({
        @SecurityScheme(
                name = "bearerAuth",
                description = "JWT Auth",
                scheme = "bearer",
                bearerFormat = "JWT",
                type = SecuritySchemeType.HTTP,
                in = SecuritySchemeIn.HEADER,
                paramName = "Authorization"
        )
})
public class OpenApiConfig {
}

File: authorization-service/src/main/kotlin/com/yazikochesalna/authorizationservice/config/OpenApiConfig.kt
package com.yazikochesalna.authorizationservice.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.util.StringUtils
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.security.SecurityScheme
import io.swagger.v3.oas.models.security.SecurityRequirement
import io.swagger.v3.oas.models.Components

@Configuration
open class OpenApiConfig(
    @Value("\${spring.application.name}")
    private val moduleName: String
) {
    private val apiVersion: String = "v2"

    @Bean
    open fun customOpenAPI(): OpenAPI {
        val securitySchemeName = "bearerAuth"
        val formattedModuleName = if (StringUtils.hasText(moduleName)) moduleName else "NoName - contact admin"
        val apiTitle = "${formattedModuleName.replaceFirstChar { it.uppercase() }} API"

        return OpenAPI()
            .addSecurityItem(SecurityRequirement().addList(securitySchemeName))
            .components(
                Components().addSecuritySchemes(
                    securitySchemeName,
                    SecurityScheme()
                        .name(securitySchemeName)
                        .type(SecurityScheme.Type.HTTP)
                        .scheme("bearer")
                        .bearerFormat("JWT")
                )
            )
            .info(Info().title(apiTitle).version(apiVersion))
    }
}